tests/functional/u/undefined/undefined_variable.py

```diff
@@ -35,7 +35,7 @@ def bad_default(var, default=unknown2):  # [undefined-variable]
 LMBD2 = lambda x, y: x+z  # [undefined-variable]
 
 try:
-    POUET # don't catch me
+    POUET # [used-before-assignment]
```

Comment:
From the blame it looks like this was tolerated just because of generic uncertainty about the control flow, but I think assuming try blocks _never_ execute is bad style.

Reply:
Does this warrant an additional entry in the changelog? I know #5384 also deals with this, but this is a change in this PR that is not covered by the changelog. I think we should add something about it.